src/Polly.Core/Registry/ResiliencePipelineRegistry.cs

```diff
@@ -258,15 +253,12 @@ private GenericRegistry<TResult> GetGenericRegistry<TResult>()
             return (GenericRegistry<TResult>)genericRegistry;
         }
 
-        return (GenericRegistry<TResult>)_genericRegistry.GetOrAdd(typeof(TResult), _ =>
-        {
-            return new GenericRegistry<TResult>(
-                () => new ResiliencePipelineBuilder<TResult>(_activator()),
-                _builderComparer,
-                _pipelineComparer,
-                _builderNameFormatter,
-                _instanceNameFormatter);
-        });
+        return (GenericRegistry<TResult>)_genericRegistry.GetOrAdd(typeof(TResult), new GenericRegistry<TResult>(
+            () => new ResiliencePipelineBuilder<TResult>(_activator()),
+            _builderComparer,
+            _pipelineComparer,
+            _builderNameFormatter,
+            _instanceNameFormatter));
```

Comment:
Same here regarding always allocating.

Reply:
This only allocates if the key wasn't already in the registry so no change from previous behavior. The delegate based `GetOrAdd` doesn't provide any additional protection against race conditions, it just saves a bit on `GetHashCode` calculations (but not in this case as `TryGetValue` was called earlier anyway).